src/mame/mr/nl_dribling.cpp

```diff
@@ -0,0 +1,595 @@
+	NET_C(IC_8A.13, IC_7A.1, IC_7A.2)
+	NET_C(IC_7A.12, IC_7B.3)
+	NET_C(R32.2, IC_7B.5, C20.2)
+	NET_C(C20.1, GND)
+	ALIAS(NOISE, IC_7A.13)
+	ALIAS(C_IN, IC_8A.8)
+
+	NET_C(I_FOLLA_B, IC_E8.10)
+	NET_C(I_FOLLA_M, IC_E8.12)
+	NET_C(I_FOLLA_A, IC_E8.1)
+	NET_C(IC_E8.9, IC_E8.13, IC_E8.2, NOISE)
+	NET_C(IC_E8.8, R33.1)
+	NET_C(IC_E8.11, R34.1)
+	NET_C(IC_E8.3, R36.1)
+	NET_C(R35.1, GND)
+	NET_C(R33.2, R34.2, R35.2, R36.2, C22.1, C21.1)
+	NET_C(C21.2, R47.2, IC_C8.14)
+	ALIAS(FOLLA, IC_C8.14)
+	NET_C(R37.2, I_V5)
+	NET_C(C22.2, R47.1, IC_C8.13)
+	NET_C(R37.1, C24.2, IC_C8.12)
+	NET_C(C24.1, GND)
+
+
+
+	// CONTRASTO
+	SUBMODEL(NE556_DIP, IC_Q9)
+	NET_C(IC_Q9.7, GND)
+	NET_C(IC_Q9.14, I_V5)
+
+	RES(R38, RES_M(1))
+	RES(R39, RES_K(100))
+	CAP(C23, CAP_U(0.1))
+	CAP(C25, CAP_U(0.1))
+	CAP(C26, CAP_U(0.47))
+	NET_C(R38.1, I_V5)
+	NET_C(R39.1, I_V5)
+	NET_C(R38.2, IC_Q9.1, IC_Q9.2, C25.1)
+	NET_C(C25.2, C26.2, GND)
+	NET_C(R39.2, IC_Q9.13, IC_Q9.12, IC_Q9.8, C26.1)
+	NET_C(IC_Q9.5, IC_Q9.10)
+	NET_C(IC_Q9.9, C23.1)
+	NET_C(IC_Q9.4, I_V5)
+	NET_C(I_CONTRASTO, IC_Q9.6)
+	ALIAS(CONT_OUT, C23.2)
+	NET_C(C23.2, R36.2)
+
+
+
+	// OUTPUT SECTION ----------------------
+	// TOS
+	CAP(C_TOS, CAP_N(100))
+	RES(R_TOS, RES_K(220))
+	NET_C(TOS, C_TOS.1)
+	NET_C(R_TOS.1, C_TOS.2)
+
+	// FISCHIO
+	CAP(C_FISCHIO, CAP_N(1000))
+	RES(R_FISCHIO, RES_K(220))
+	NET_C(FISCHIO, C_FISCHIO.1)
+	NET_C(R_FISCHIO.1, C_FISCHIO.2)
+
+	// FOLLA
+	CAP(C_FOLLA, CAP_N(10))
+	RES(R_FOLLA, RES_K(100))
+	NET_C(FOLLA, C_FOLLA.1)
+	NET_C(C_FOLLA.2, R_FOLLA.1)
+
+	// CALCIO_A
+	CAP(C_CALCIO_A, CAP_U(1))
+	RES(R_CALCIO_A, RES_K(10))
+	NET_C(CALCIO_A.OUTPUT, C_CALCIO_A.1)
+	NET_C(C_CALCIO_A.2, R_CALCIO_A.1)
+
+	// CALCIO_B
+	CAP(C_CALCIO_B, CAP_U(1))
+	RES(R_CALCIO_B, RES_K(10))
+	NET_C(CALCIO_B.OUTPUT, C_CALCIO_B.1)
+	NET_C(C_CALCIO_B.2, R_CALCIO_B.1)
+
+	// STOP_PALLA
+	CAP(C_STOP_PALLA, CAP_U(1))
+	RES(R_STOP_PALLA, RES_K(100))
+	NET_C(STOP_PALLA.OUTPUT, C_STOP_PALLA.1)
+	NET_C(C_STOP_PALLA.2, R_STOP_PALLA.1)
+
+	// PARATA
+	CAP(C_PARATA, CAP_U(1))
+	RES(R_PARATA, RES_K(100))
+	NET_C(PARATA.OUTPUT, C_PARATA.1)
+	NET_C(C_PARATA.2, R_PARATA.1)
+
+	// FINAL MIX AND DISABLE CIRCUITRY
+	TTL_INPUT(ENABLE_SOUND, 0)
+	NET_C(ENABLE_SOUND.VCC, I_V5)
+	NET_C(ENABLE_SOUND.GND, GND)
+	QBJT_EB(Q_OUT, "BC239C")
+	RES(R_OUT, RES_K(2.2))
+	RES(R_PU_OUT, RES_K(10))
+	NET_C(R_PU_OUT.1, I_V5)
+	NET_C(ENABLE_SOUND, R_OUT.1, R_PU_OUT.2)
+	NET_C(Q_OUT.B, R_OUT.2)
+	NET_C(Q_OUT.C, R_TOS.2, R_FISCHIO.2, R_FOLLA.2, R_CALCIO_A.2, R_CALCIO_B.2, R_STOP_PALLA.2, R_PARATA.2)
+	NET_C(Q_OUT.E, GND)
+
+	// 3rd order Butterworth filter approximation.
+	SUBMODEL(output_filter, OUTPUT_FILTER)
+	SUBMODEL(output_filter, OUTPUT_FILTER_2)
+	SUBMODEL(output_filter, OUTPUT_FILTER_3)
+	NET_C(OUTPUT_FILTER.INPUT, Q_OUT.C)
+	NET_C(OUTPUT_FILTER.GND,OUTPUT_FILTER_2.GND, OUTPUT_FILTER_3.GND, GND)
+	NET_C(OUTPUT_FILTER.OUTPUT, OUTPUT_FILTER_2.INPUT)
+	NET_C(OUTPUT_FILTER_2.OUTPUT, OUTPUT_FILTER_3.INPUT)
```

Comment:
24kHz corner frequency is too high.  For a 48kHz sample rate, the Nyquist frequency is 24kHz.  You want anything above this to be suppressed significantly.  This isn’t going to happen if you have the corner frequency right at the Nyquist frequency.  The TDA2003 has a cutoff frequency of 15kHz – that might be a better choice.

Also, this is an over-complicated way of doing this.  You don’t need to cascade three first-order filters.  It’s more efficient to do it in a single stage with the Sallen-Key arrangement.

For example there’s a tool to calculate parameters here: http://sim.okawa-denshi.jp/en/Sallen3tool.php

For example you can select a Butterworth filter approximation and enter 15000 for the frequency and see what it produces.

And why are you using an LM324 in each stage of your output filter in the first place?  You only need one op-amp, not four, and I suggested using an idealised op-amp model, as the purpose is just to suppress high-frequency components that would otherwise cause aliasing.

For example you could try `OPAMP(TYPE=1 FPF=5 RI=1M RO=50 UGF=1M)` as a starting point.

Reply:
> The TDA2003 has a cutoff frequency of 15kHz – that might be a better choice.

This is (seemingly, unless I'm misreading the datasheet?) only true if the Cx and Rx components are populated, and in the case of Dribbling, this doesn't seem to be the case. I'm not sure if the TDA2003 circuit has any specific filter cutoff at all, without those components. Again, I could be wrong.

Reply:
Confirmed that this works:

```
    // 15KHz low-pass filter
    OPAMP(AMP, "OPAMP(TYPE=1 FPF=5 RI=1M RO=50 UGF=1M SLEW=1M VLH=0.5 VLL=0.03 DAB=0.0015)")
```

I had to populate more parameters, but I used the ones from the example and it sounds well. But I am no expert.

Reply:
> > The TDA2003 has a cutoff frequency of 15kHz – that might be a better choice.
> 
> This is (seemingly, unless I'm misreading the datasheet?) only true if the Cx and Rx components are populated, and in the case of Dribbling, this doesn't seem to be the case. I'm not sure if the TDA2003 circuit has any specific filter cutoff at all, without those components. Again, I could be wrong.

Nah, 15kHz is the typical -3dB frequency with the recommended R1/R2/R3/C2/C5, which are all included on the Dribbling schematic.  You can add Rx and Cx to lower the upper cutoff frequency

The formula they give is Cx=1÷(2×π×B×R1) so B=1÷(2×π×Cx×R1).  With the test circuit values of R1=220Ω and Cx=39nF, you get B=18.5kHz.  It adds additional first-order roll-off above 18kHz, but it doesn’t change the -3dB point.

Reply:
> > The TDA2003 has a cutoff frequency of 15kHz – that might be a better choice.
> 
> This is (seemingly, unless I'm misreading the datasheet?) only true if the Cx and Rx components are populated, and in the case of Dribbling, this doesn't seem to be the case. I'm not sure if the TDA2003 circuit has any specific filter cutoff at all, without those components. Again, I could be wrong.

The filter fixed the aliasing which is an emulator issue only though. Not really replacing the TDA2003 in strict sense.